ArcadiaTeamsBot.ServiceDesk/ServiceDeskClient.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace ArcadiaTeamsBot.ServiceDesk
+{
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using ArcadiaTeamsBot.ServiceDesk.Abstractions;
+    using ArcadiaTeamsBot.ServiceDesk.Abstractions.DTOs;
+
+    public class ServiceDeskClient : IServiceDeskClient
+    {
+        private const string requestTypesUrl = "intra/requestTypes";
+        private const string currentRequestsUrl = "intra/requests?username=";
+
+        private readonly IHttpClientFactory clientFactory;
+
+        private readonly IEnumerable<ServiceDeskRequestPriorityDTO> priorities = new[]
+        {
+            new ServiceDeskRequestPriorityDTO
+            {
+                Key = 1,
+                Value = "Low"
+            },
+            new ServiceDeskRequestPriorityDTO
+            {
+                Key = 2,
+                Value = "Default"
+            },
+            new ServiceDeskRequestPriorityDTO
+            {
+                Key = 3,
+                Value = "High"
+            }
+        };
+
+        private readonly ServiceDeskConfiguration serviceDeskConfiguration;
+
+        public ServiceDeskClient(IHttpClientFactory clientFactory, ServiceDeskConfiguration serviceDeskConfiguration)
+        {
+            this.clientFactory = clientFactory;
+            this.serviceDeskConfiguration = serviceDeskConfiguration;
+        }
+
+        public Task<IEnumerable<ServiceDeskRequestTypeDTO>> GetRequestTypes(CancellationToken cancellationToken)
+        {
+            return this.GetByUrl<IEnumerable<ServiceDeskRequestTypeDTO>>($"{this.serviceDeskConfiguration.ApiUrl}{requestTypesUrl}", cancellationToken);
+        }
+
+        public Task<IEnumerable<ServiceDeskRequestDTO>> GetCurrentRequests(string username, CancellationToken cancellationToken)
+        {
+            return this.GetByUrl<IEnumerable<ServiceDeskRequestDTO>>($"{this.serviceDeskConfiguration.ApiUrl}{currentRequestsUrl}{username}", cancellationToken);
+        }
+
+        public Task<IEnumerable<ServiceDeskRequestPriorityDTO>> GetPriorities()
+        {
+            return Task.FromResult(this.priorities);
+        }
+
+        public async Task<T> GetByUrl<T>(string url, CancellationToken cancellationToken)
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, url);
+
+            foreach (var header in this.serviceDeskConfiguration.Headers)
```

Comment:
I think it is enough to have only `ApiKey` property in application settings for Service Desk and apply it to `x-api-key` header. Headers list for all possible use-cases is a kind of execssive overhead.

Reply:
Made it one ApiKey prop